lib/msal-browser/src/telemetry/BrowserPerformanceClient.ts

```diff
@@ -56,4 +56,53 @@ export class BrowserPerformanceClient extends PerformanceClient implements IPerf
             }
         };
     }
+
+    /**
+     * Adds pre-queue time to preQueueTimeByCorrelationId map.
+     * @param {PerformanceEvents} eventName 
+     * @param {?string} correlationId 
+     * @returns 
+     */
+    setPreQueueTime(eventName: PerformanceEvents, correlationId?: string): void {
+        if (!correlationId) {
+            this.logger.trace(`BrowserPerformanceClient: correlationId for ${eventName} not provided, unable to set telemetry queue time`);
+            return;
+        }
+
+        const preQueueTimesByEvents = this.preQueueTimeByCorrelationId.get(correlationId);
+
+        if (preQueueTimesByEvents){
+            preQueueTimesByEvents.set(eventName, window.performance.now());
```

Comment:
Do we need to check that `window` and `window.performance` are available? Or is that done somewhere else?

Reply:
Thanks for pointing that out. Added checks for window performance API.